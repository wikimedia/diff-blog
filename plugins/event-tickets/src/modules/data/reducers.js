/**
 * External dependencies
 */
import { combineReducers } from 'redux';

/**
 * Internal dependencies
 */
import blocks from './blocks';
import move from './shared/move/reducers';

export default combineReducers( {
	blocks,
	move,
} );
