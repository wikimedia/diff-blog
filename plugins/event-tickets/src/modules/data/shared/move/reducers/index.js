/**
 * External dependencies
 */
import { combineReducers } from 'redux';

/**
 * Internal dependencies
 */
import posts from './posts';
import postTypes from './postTypes';
import ui from './ui';
import modal from './modal';

export default combineReducers( {
	posts,
	postTypes,
	ui,
	modal,
} );
