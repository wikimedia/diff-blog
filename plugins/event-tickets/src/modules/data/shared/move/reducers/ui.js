/**
 * Internal dependencies
 */
import * as types from '@moderntribe/tickets/data/shared/move/types';

export const DEFAULT_STATE = {
	showModal: false,
};

export default function ui( state = DEFAULT_STATE, action ) {
	switch ( action.type ) {
		case types.SHOW_MODAL:
			return {
				...state,
				showModal: true,
			};
		case types.HIDE_MODAL:
			return {
				...state,
				showModal: false,
			};
		default:
			return state;
	}
}

