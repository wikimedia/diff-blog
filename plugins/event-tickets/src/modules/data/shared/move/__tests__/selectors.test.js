/**
 * External Dependencies
 */
import * as selectors from '../selectors';
import { DEFAULT_STATE as MODAL_DEFAULT_STATE } from '../reducers/modal';
import { DEFAULT_STATE as POSTS_DEFAULT_STATE } from '../reducers/posts';
import { DEFAULT_STATE as POST_TYPES_DEFAULT_STATE } from '../reducers/postTypes';
import { DEFAULT_STATE as UI_DEFAULT_STATE } from '../reducers/ui';

describe( 'Move Selectors', () => {
	let state;

	beforeEach( () => {
		state = {
			tickets: {
				move: {
					ui: Object.assign( {}, UI_DEFAULT_STATE ),
					posts: Object.assign( {}, POSTS_DEFAULT_STATE ),
					modal: Object.assign( {}, MODAL_DEFAULT_STATE ),
					postTypes: Object.assign( {}, POST_TYPES_DEFAULT_STATE ),
				},
			},
		};
	} );

	Object.keys( selectors ).forEach( ( key ) => {
		test( `Default - ${ key }`, () => {
			expect( selectors[ key ]( state ) ).toMatchSnapshot();
		} );
	} );

	test( 'getPostTypeOptions', () => {
		state.tickets.move.postTypes.posts = {
			all: 'All',
			cool: 'Cool',
		};
		expect( selectors.getPostTypeOptions( state ) ).toMatchSnapshot();
	} );

	test( 'getPostOptions', () => {
		state.tickets.move.postTypes.posts = {
			all: 'All',
			cool: 'Cool',
		};
		expect( selectors.getPostOptions( state ) ).toMatchSnapshot();
	} );

	test( 'getPostTypeOptionValue', () => {
		state.tickets.move.postTypes.posts = {
			all: 'All',
			cool: 'Cool',
		};
		state.tickets.move.modal.post_type = 'all';
		expect( selectors.getPostTypeOptionValue( state ) ).toMatchSnapshot();
	} );
	test( 'hasSelectedPost', () => {
		state.tickets.move.posts.posts = {
			all: 'All',
			cool: 'Cool',
		};
		state.tickets.move.modal.target_post_id = 'all';
		expect( selectors.hasSelectedPost( state ) ).toMatchSnapshot();
	} );
} );
