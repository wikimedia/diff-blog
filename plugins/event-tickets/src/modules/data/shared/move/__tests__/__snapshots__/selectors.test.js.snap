// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Move Selectors Default - _getModal 1`] = `
Object {
  "clientId": null,
  "isSubmitting": false,
  "post_type": "all",
  "search_terms": "",
  "target_post_id": null,
  "ticketId": null,
}
`;

exports[`Move Selectors Default - _getPostTypes 1`] = `
Object {
  "isFetching": false,
  "posts": Object {},
}
`;

exports[`Move Selectors Default - _getPosts 1`] = `
Object {
  "isFetching": false,
  "posts": Object {},
}
`;

exports[`Move Selectors Default - _getUI 1`] = `
Object {
  "showModal": false,
}
`;

exports[`Move Selectors Default - getModalClientId 1`] = `null`;

exports[`Move Selectors Default - getModalPostType 1`] = `"all"`;

exports[`Move Selectors Default - getModalSearch 1`] = `""`;

exports[`Move Selectors Default - getModalTarget 1`] = `null`;

exports[`Move Selectors Default - getModalTicketId 1`] = `null`;

exports[`Move Selectors Default - getMove 1`] = `
Object {
  "modal": Object {
    "clientId": null,
    "isSubmitting": false,
    "post_type": "all",
    "search_terms": "",
    "target_post_id": null,
    "ticketId": null,
  },
  "postTypes": Object {
    "isFetching": false,
    "posts": Object {},
  },
  "posts": Object {
    "isFetching": false,
    "posts": Object {},
  },
  "ui": Object {
    "showModal": false,
  },
}
`;

exports[`Move Selectors Default - getPostOptions 1`] = `Array []`;

exports[`Move Selectors Default - getPostTypeOptionValue 1`] = `undefined`;

exports[`Move Selectors Default - getPostTypeOptions 1`] = `Array []`;

exports[`Move Selectors Default - getPostTypes 1`] = `Object {}`;

exports[`Move Selectors Default - getPosts 1`] = `Object {}`;

exports[`Move Selectors Default - hasSelectedPost 1`] = `false`;

exports[`Move Selectors Default - isFetchingPostTypes 1`] = `false`;

exports[`Move Selectors Default - isFetchingPosts 1`] = `false`;

exports[`Move Selectors Default - isModalShowing 1`] = `false`;

exports[`Move Selectors Default - isModalSubmitting 1`] = `false`;

exports[`Move Selectors getPostOptions 1`] = `Array []`;

exports[`Move Selectors getPostTypeOptionValue 1`] = `
Object {
  "label": "All",
  "value": "all",
}
`;

exports[`Move Selectors getPostTypeOptions 1`] = `
Array [
  Object {
    "label": "All",
    "value": "all",
  },
  Object {
    "label": "Cool",
    "value": "cool",
  },
]
`;

exports[`Move Selectors hasSelectedPost 1`] = `true`;
